src/app/billing-client.ts

```diff
@@ -0,0 +1,117 @@
+import { type RpcOptions } from '@improbable-eng/grpc-web/dist/typings/client.d';
+import { BillingServiceClient } from '../gen/app/v1/billing_pb_service';
+import pb from '../gen/app/v1/billing_pb';
+import { promisify } from '../utils';
+
+type GetCurrentMonthUsageResponse =
+  Partial<pb.GetCurrentMonthUsageResponse.AsObject> & {
+    start?: Date;
+    end?: Date;
+  };
```

Comment:
(q) pardon my typescript ignorance, why is this necessary? Can we not set `start` or `end` on a `GetCurrentMonthUsageResponse` object without doing this?

Reply:
Good question! We have to define the properties in Typescript. Imo, it also helps users to figure out what the actual property is instead of adding it without any mention outside of the function itself.